pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1265,26 +1281,53 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 
 	volumePlugin, err := og.volumePluginMgr.FindExpandablePluginBySpec(volumeSpec)
 
-	if err != nil {
+	if err != nil && volumeSpec.IsControllerExpandable() {
 		return volumetypes.GeneratedOperations{}, fmt.Errorf("Error finding plugin for expanding volume: %q with error %v", pvcWithResizeRequest.QualifiedName(), err)
+	} else if !volumeSpec.IsControllerExpandable() {
+		// A flex volume is expandable but since it resides on the kubelet not the controller
+		// and is dynamically probed plugin, FindExpandablePluginBySpec will never find it
+		// Therefore fake the volume expansion on the controller.
+		glog.V(4).Infof("Faking expansion for flex volume %s", pvcWithResizeRequest.QualifiedName())
 	}
-	if volumePlugin == nil {
+
+	if volumePlugin == nil && !volumeSpec.IsControllerExpandable() {
 		return volumetypes.GeneratedOperations{}, fmt.Errorf("Can not find plugin for expanding volume: %q", pvcWithResizeRequest.QualifiedName())
 	}
 
+	getPluginName := func() string {
+		if volumePlugin != nil {
+			return volumePlugin.GetPluginName()
+		}
+		if !volumeSpec.IsControllerExpandable() {
+			return volumeSpec.NonControllerExpandablePluginName()
+		}
+		return ""
+	}
+
 	expandVolumeFunc := func() (error, error) {
 		newSize := pvcWithResizeRequest.ExpectedSize
 		pvSize := pvcWithResizeRequest.PersistentVolume.Spec.Capacity[v1.ResourceStorage]
 		if pvSize.Cmp(newSize) < 0 {
-			updatedSize, expandErr := volumePlugin.ExpandVolumeDevice(
-				volumeSpec,
-				pvcWithResizeRequest.ExpectedSize,
-				pvcWithResizeRequest.CurrentSize)
+			glog.Warningf("calling ExpandVolume for volume %s", pvcWithResizeRequest.QualifiedName())
 
-			if expandErr != nil {
-				detailedErr := fmt.Errorf("Error expanding volume %q of plugin %s : %v", pvcWithResizeRequest.QualifiedName(), volumePlugin.GetPluginName(), expandErr)
-				return detailedErr, detailedErr
+			var updatedSize resource.Quantity
+
+			if volumePlugin != nil {
+				var expandErr error
+				updatedSize, expandErr = volumePlugin.ExpandVolumeDevice(
+					volumeSpec,
+					pvcWithResizeRequest.ExpectedSize,
+					pvcWithResizeRequest.CurrentSize)
+
+				if expandErr != nil {
+					detailedErr := fmt.Errorf("error expanding volume %q of plugin %q: %v", pvcWithResizeRequest.QualifiedName(), volumePlugin.GetPluginName(), expandErr)
+					return detailedErr, detailedErr
+				}
+			} else {
+				// this is the case for flex volume, which doesn't reside on the controller
```

Comment:
Same here - Change the comment to "if plugin is not found, assume it is resizable"

Reply:
OK